### PR TITLE
minor ux improvements for scm

### PR DESCRIPTION
**Pull request by GnsP**
# minor ux improvements for scm

## Description
Improves data fetching in the scm sync tab, and some copy changes.
Contains a subset of changes from #1157 , as the former PR got larger and not all changes there are cleared to be included in the release 6.10.0

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [x] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-20917](https://cdap.atlassian.net/browse/CDAP-20917)

## Test Plan
e2e tests added already.

## Screenshots
NA

[CDAP-20917]: https://cdap.atlassian.net/browse/CDAP-20917?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/cdap/components/SourceControlManagement/helpers.ts [hunk 1]
```diff
@@ -123,3 +143,22 @@ export const compareTimeInstant = (t1: ITimeInstant, t2: ITimeInstant): number =
...
+  }
+  return null;
+};
+
+export const timeInstantToString = (t?: ITimeInstant): string => {
+  if (!t) {
+    return '--';
+  }
+
+  return moment(t.seconds * 1000).format('DD-MM-YYYY HH:mm:ss A');
```
Comment (njbriggs) [inline app/cdap/components/SourceControlManagement/helpers.ts]: For future consideration: consolidate with date formatting logic in https://github.com/cdapio/cdap-ui/blob/develop/app/cdap/services/DataFormatter/index.ts

Comment (GnsP) [reply app/cdap/components/SourceControlManagement/helpers.ts]: done.

File: app/cdap/components/SourceControlManagement/OperationAlert.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    const timeTaken = getOperationRunTime(operation);
+    if (!timeTaken) {
+      return `${startedPrefix} ${startTime}`;
+    }
+
+    const timeTakenPrefix = operation.done
+      ? T.translate(`${PREFIX}.operationRanFor`).toString()
+      : T.translate(`${PREFIX}.operationRunningFor`).toString();
+
+    return `${startedPrefix} ${startTime}, ${timeTakenPrefix} ${timeTaken}`;
```
Comment (njbriggs) [inline app/cdap/components/SourceControlManagement/OperationAlert.tsx]: We should not be using template literals to generate localized strings - some languages will expect segments to be in a different order. To be fully internationalized, this should be done through `T.translate` (L102 also)

**Review by njbriggs — APPROVED** [review]
One question about i18n - up to you if you want to address it.

File: app/cdap/components/SourceControlManagement/OperationAlert.tsx (hunk 2 above)
Comment (GnsP) [reply app/cdap/components/SourceControlManagement/OperationAlert.tsx]: Agreed. This current implementation may not work properly in all cases, especially in RTL languages. Will update it.

**Review by sumengwang — APPROVED** [review]

File: app/cdap/components/SourceControlManagement/OperationAlert.tsx (hunk 2 above)
Comment (GnsP) [reply app/cdap/components/SourceControlManagement/OperationAlert.tsx]: Updated.

**Pull request merged**